src/Context.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useMemo, useState } from 'react';
+import { Post } from './types/interfaces';
+
+interface IAppContext {
+  isModalWindowShown: boolean;
+  toggleOpen?: (value: boolean) => void;
+  modal: string;
+  toggleModal?: (value: string) => void;
+  editedPost?: Post | null;
+  setEditedPostVal?: (value: Post) => void;
+}
+
+interface ChildrenProps {
+  children: React.ReactNode;
+}
+
+const defaultState: IAppContext = {
+  isModalWindowShown: false,
+  modal: '',
+  editedPost: null,
+};
+
+export const AppContext = React.createContext<IAppContext>(defaultState);
+
+export const AppProvider = ({ children }: ChildrenProps) => {
+  const [isOpen, setIsOpen] = useState(defaultState.isModalWindowShown);
```

Comment:
If you follow my recommendations you won't need this state, you can check just modalType